src/business_logic/transaction/objects/internal_invoke_function.rs

```diff
@@ -418,4 +418,257 @@ mod tests {
         );
         assert_eq!(result.call_info.unwrap().retdata, vec![Felt252::new(144)]);
     }
+
+    #[test]
+    fn test_execute_specific_concurrent_changes() {
+        let internal_invoke_function = InternalInvokeFunction {
+            contract_address: Address(0.into()),
+            entry_point_selector: Felt252::from_str_radix(
+                "112e35f48499939272000bd72eb840e502ca4c3aefa8800992e8defb746e0c9",
+                16,
+            )
+            .unwrap(),
+            entry_point_type: EntryPointType::External,
+            calldata: vec![1.into(), 1.into(), 10.into()],
+            tx_type: TransactionType::InvokeFunction,
+            version: 0,
+            validate_entry_point_selector: 0.into(),
+            hash_value: 0.into(),
+            signature: Vec::new(),
+            max_fee: 0,
+            nonce: Some(0.into()),
+        };
+
+        // Instantiate CachedState
+        let mut state_reader = InMemoryStateReader::default();
+        // Set contract_class
+        let class_hash = [1; 32];
+        let contract_class =
+            ContractClass::try_from(PathBuf::from("starknet_programs/fibonacci.json")).unwrap();
+        // Set contact_state
+        let contract_address = Address(0.into());
+        let nonce = Felt252::zero();
+
+        state_reader
+            .address_to_class_hash_mut()
+            .insert(contract_address.clone(), class_hash);
+        state_reader
+            .address_to_nonce
+            .insert(contract_address, nonce);
+
+        let mut state = CachedState::new(state_reader.clone(), None);
+
+        // Initialize state.contract_classes
+        state.set_contract_classes(HashMap::new()).unwrap();
+
+        state
+            .set_contract_class(&class_hash, &contract_class)
+            .unwrap();
+
+        let result = internal_invoke_function
+            .execute(&mut state, &StarknetGeneralConfig::default())
+            .unwrap();
+
+        assert_eq!(result.tx_type, Some(TransactionType::InvokeFunction));
+        assert_eq!(
+            result.call_info.as_ref().unwrap().class_hash,
+            Some(class_hash)
+        );
+        assert_eq!(
+            result.call_info.as_ref().unwrap().entry_point_selector,
+            Some(internal_invoke_function.entry_point_selector)
+        );
+        assert_eq!(
+            result.call_info.as_ref().unwrap().calldata,
+            internal_invoke_function.calldata
+        );
+        assert_eq!(result.call_info.unwrap().retdata, vec![Felt252::new(144)]);
+    }
+
+    #[test]
+    fn test_execute_invoke_max_fee_non_zero_should_fail() {
```

Comment:
We should rename this test, as `max_fee` can be non-zero and should not fail because of that. Now it does, but only because fee calculation is a WIP. (pls correct me if I'm wrong)

Reply:
It's true that's is WIP but just checked and it should actually be named after a different error. It fails with max_fee != 0 because then it does the whole fee calculation and the Starknet config default does not support all the information required for the calculation. Changing the test name to express this correctly. Thanks for making me aware of this!

Reply:
It's true that's is WIP but just checked and it should actually be named after a different error. It fails with max_fee != 0 because then it does the whole fee calculation and the Starknet config default does not support all the information required for the calculation. Changing the test name to express this correctly. Thanks for making me aware of this!